server/src/main/java/com/linecorp/centraldogma/server/internal/api/sysadmin/KeyManagementService.java

```diff
@@ -0,0 +1,117 @@
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import com.linecorp.armeria.server.annotation.Blocking;
+import com.linecorp.armeria.server.annotation.Default;
+import com.linecorp.armeria.server.annotation.Get;
+import com.linecorp.armeria.server.annotation.Param;
+import com.linecorp.armeria.server.annotation.Post;
+import com.linecorp.armeria.server.annotation.ProducesJson;
+import com.linecorp.centraldogma.server.auth.SessionMasterKey;
+import com.linecorp.centraldogma.server.command.Command;
+import com.linecorp.centraldogma.server.command.CommandExecutor;
+import com.linecorp.centraldogma.server.internal.api.AbstractService;
+import com.linecorp.centraldogma.server.internal.api.auth.RequiresSystemAdministrator;
+import com.linecorp.centraldogma.server.storage.encryption.EncryptionStorageManager;
+import com.linecorp.centraldogma.server.storage.encryption.SecretKeyWithVersion;
+import com.linecorp.centraldogma.server.storage.encryption.WrappedDekDetails;
+
+@ProducesJson
+@RequiresSystemAdministrator
+public final class KeyManagementService extends AbstractService {
+
+    private final EncryptionStorageManager encryptionStorageManager;
+
+    public KeyManagementService(CommandExecutor executor, EncryptionStorageManager encryptionStorageManager) {
+        super(executor);
+        requireNonNull(encryptionStorageManager, "encryptionStorageManager");
+        checkArgument(encryptionStorageManager.enabled(), "EncryptionStorageManager must be enabled.");
+        this.encryptionStorageManager = encryptionStorageManager;
+    }
+
+    @Get("/wdeks")
+    @Blocking
+    public List<WrappedDekDetails> listWdeks() {
+        return encryptionStorageManager.wdeks();
+    }
+
+    /**
+     * Rotates the wrapped data encryption key (WDEK) for the specified project and repository.
+     */
+    @Post("/projects/{projectName}/repos/{repoName}/wdeks/rotate")
+    @Blocking
+    public CompletableFuture<Void> rotateWdek(@Param String projectName, @Param String repoName,
+                                              @Param @Default("false") boolean reencrypt) {
+        requireNonNull(projectName, "projectName");
+        requireNonNull(repoName, "repoName");
+        if (reencrypt) {
+            // TODO(minwoox): Implement re-encryption logic.
+            throw new UnsupportedOperationException();
+        } else {
+            final SecretKeyWithVersion currentDek =
+                    encryptionStorageManager.getCurrentDek(projectName, repoName);
+            return encryptionStorageManager.generateWdek().thenCompose(wdek -> {
+                // If concurrent rotation happened, only one of them will succeed to create a new WDEK because
+                // the storage manager accepts only the WDEK with the version which is exactly one greater.
```

Comment:
Question) If the same command is accidentally submitted to two different replicas and both succeed locally, could the ZooKeeper replication be stopped?